tools/packaging/packaging.psm1

```diff
@@ -886,6 +742,324 @@ function New-UnixPackage {
     }
 }
 
+
+function Get-FpmArguments
+{
+    param(
+        [Parameter(Mandatory,HelpMessage='Package Name')]
+        [String]$Name,
+
+        [Parameter(Mandatory,HelpMessage='Package Version')]
+        [String]$Version,
+
+        [Parameter(Mandatory)]
+        [String]$Iteration,
+
+        [Parameter(Mandatory,HelpMessage='Package description')]
+        [String]$Description,
+
+        [Parameter(Mandatory,HelpMessage='Installer Type')]
+        [String]$Type,
+
+        [Parameter(Mandatory,HelpMessage='Staging folder for installation files')]
+        [String]$Staging,
+
+        [Parameter(Mandatory,HelpMessage='Install path on target machine')]
+        [String]$Destination,
+
+        [Parameter(Mandatory,HelpMessage='The built and gzipped mann file.')]
+        [String]$MannGzipFile,
+
+        [Parameter(Mandatory,HelpMessage='The destination of the mann file')]
+        [String]$MannDestination,
+
+        [Parameter(Mandatory,HelpMessage='Symlink to powershell executable')]
+        [String]$LinkSource,
+
+        [Parameter(Mandatory,HelpMessage='Destination for symlink to powershell executable')]
+        [String]$LinkDestination,
+
+        [Parameter(HelpMessage='Packages required to install this package.  Not applicable for MacOS.')]
+        [ValidateScript({
+            if (!$Environment.IsMacOS -and $_.Count -eq 0)
+            {
+                throw "Must not be null or empty on this environment."
+            }
+            return $true
+        })]        [String[]]$Dependencies,
+
+        [Parameter(HelpMessage='Script to run after the package installation.')]
+        [AllowNull()]
+        [ValidateScript({
+            if (!$Environment.IsMacOS -and !$_)
+            {
+                throw "Must not be null on this environment."
+            }
+            return $true
+        })]
+        [String]$AfterInstallScript,
+
+        [Parameter(HelpMessage='Script to run after the package removal.')]
+        [AllowNull()]
+        [ValidateScript({
+            if (!$Environment.IsMacOS -and !$_)
+            {
+                throw "Must not be null on this environment."
+            }
+            return $true
+        })]
+        [String]$AfterRemoveScript
+    )
+
+    $Arguments = @(
```

Comment:
Consider ArrayList

Reply:
resolved

Reply:
This became unreadable after changing to an array list.  I filed this issue to add a feature to PowerShell so I can maintain a similar syntax and use a list:
https://github.com/PowerShell/PowerShell/issues/5643